Met à jour la doc pour Python 3

motet-a opened the pull request:
Commit f8ecfee de @pierre-24, que j’ai tiré de la PR #4503.

pierre-24 committed:
Met à jour la doc pour Python 3

Commit f8ecfeee3c8d60d847502a173a758e031ef6e168 de pierre_24, que j’ai
tiré de la PR #4503.

motet-a committed:
Met à jour la page “À propos” pour Python 3

Je n’ai pas mis le numéro de version mineur puisque ça risque de
changer de temps en temps, c’est toujours pénible à mettre à jour et
les utilisateurs s’en fichent probablement.

pquentin commented on doc/source/install/backend-windows-install.rst:
```diff
@@ -37,12 +37,12 @@ Prérequis
     - `Git <http://git-scm.com/download/win>`_ (Git pour Eclipse ne suffit pas ; associez les .sh).
     - `gettext <https://www.gnu.org/software/gettext/>`_.
-- `Téléchargez et installez Python 2.7 <https://www.python.org/download/releases/2.7/>`_.
+- `Téléchargez et installez Python 3.4 <https://www.python.org/downloads/release/python-344/>`_.
```
La version 3.4.6 est sortie près d'un an après : https://www.python.org/downloads/release/python-346/

motet-a replied:
Tiens, d’ailleurs il y a même [3.4.7](https://www.python.org/downloads/release/python-347/).

motet-a replied:
Par contre, c’est indiqué « Also, Python 3.4.6 has only been released in source code form ».

pquentin replied:
Du coup ouais 3.4.7 c'est mieux !

motet-a replied:
Ouais mais visiblement il n’y a pas de binaires, il faut tout compiler sois-même, non ? Et il faut peut-être aussi faire attention à la version de Python sur la production et sur la CI.

(Après je ne connais pas énormément Python donc bon)

pquentin replied:
OK je pensais que 3.4.7 avait des binaires mais pas 3.4.6. Du coup ouais, https://www.python.org/downloads/release/python-344/. Mais ce serait bien de passer à une version plus récente.

Situphen commented on doc/source/install/backend-linux-install.rst:
```diff
@@ -16,7 +16,7 @@ Certaines des commandes d'installation (débutant par ``apt-get``) sont données
 - git : ``apt-get install git``
-- python2.7
+- python3.4
 - python-dev : ``apt-get install python-dev``
```
`python3-dev` non ?

motet-a replied:
Je crois que le nom dépend des distros. C’est `python3-dev` sous Debian/Ubuntu, mais il me semble que sous Arch par exemple, ils ont tendance à appeller Python 3 simplement `python`.

Situphen commented:
Il faut penser à modifier le contenu du Makefile car il y a encore les paquets de Python 2 !

motet-a commented:
D’ailleurs, [voici les paquets à installer sous Debian Stretch pour Python 3](https://github.com/motet-a/zds-site/blob/7c5040808b37cfa7564eda69490821b8dea84fcf/Dockerfile_back#L5-L8). Attention, c’est testé uniquement sous Debian Stretch, je ne sais pas si ça marche sur Debian Stable ou Ubuntu ou quoi que ce soit d’autre. Notez que je ne sais pas si c’est une bonne idée d’installer tout ça si vous avez un venv (je n’utilise jamais de venv avec Docker).

`xvfb` et `xauth` ne sont utiles que si vous voulez passer les tests avec Selenium sans interface graphique (en gros c’est inutile). `firefox-esr` n’est utile que si vous voulez passer les tests avec Selenium.

motet-a committed:
doc : Mise-à-jour des paquets à installer pour Python 3

Attention, c’est testé sur Debian uniquement.

Je ne suis pas sûr concernant `easy_install` (je ne me suis jamais
servi de `easy_install`).

artragis commented on doc/source/install/backend-linux-install.rst:
```diff
@@ -65,8 +62,8 @@ Installation et configuration de `virtualenv`
 
-    pip install --user virtualenv # Ajout du module virtualenv
-    virtualenv zdsenv --python=python2 # Création du répertoire "zdsenv"
+    pip3 install --user virtualenv # Ajout du module virtualenv
```
Pas utile le `pip3` ici.

pquentin replied:
En vrai il faudrait utiliser `python3 -m venv zdsenv`, c'est beaucoup plus propre et plus rapide à l'utilisation que virtualenv.

gllmc replied:
Bonne idée en effet, à noter que sur Ubuntu, ça implique d'installer le paquet `python3-venv`.

motet-a replied:
Du coup, vous remplaceriez `pip3 install --user virtualenv` par `python3 -m venv zdsenv` en conseillant d’installer (au préalable) `python3-venv` ?

gllmc replied:
Je pense que ça peut se faire, mais il faut gérer les différentes distributions (je ne sais pas comment ça se fait pour les autres). Sinon, ça peut se faire plus tard (dans ce cas, laisse `pip` pour l'utilisation de `virtualenv`).

motet-a replied:
OK, donc je remplace juste `pip3` par `pip` ?

pquentin replied:
Ouais.

philae-ael commented on doc/source/install/backend-linux-install.rst:
```diff
@@ -16,22 +16,19 @@ Certaines des commandes d'installation (débutant par ``apt-get``) sont données
-- tox : ``pip install tox``
+- python3 et python3-dev : ``apt-get install python3-dev``
+- setuptools : ``apt-get install python3-setuptools``
+- pip : ``apt-get install pip3``
```
Sous [Debian](https://packages.debian.org/search?keywords=python3-pip&searchon=names&suite=stable&section=all) comme sous [Ubuntu](https://packages.ubuntu.com/search?mode=exactfilename&suite=zesty&section=all&arch=any&keywords=python3-pip&searchon=names), ça semble être python3-pip

motet-a replied:
Oh oui, bien vu. Je me suis emmêlé les pinceaux.

motet-a committed:
Modifications de @AmarOk1412

Voir #4691.

motet-a committed:
fixup! Met à jour la doc pour Python 3
